apps/prairielearn/src/ee/lib/contextEmbeddings.ts

```diff
@@ -0,0 +1,60 @@
+import { OpenAI } from 'openai';
+
+import { loadSqlEquiv, queryRows, queryOptionalRow } from '@prairielearn/postgres';
+
+import { QuestionGenerationContextEmbeddingSchema } from '../../lib/db-types.js';
+import { ServerJob } from '../../lib/server-jobs.js';
+
+import { DocumentChunk } from './context-parsers/documentation.js';
+
+const sql = loadSqlEquiv(import.meta.url);
+
+export function vectorToString(vec: number[]) {
+  return `[${vec.join(', ')}]`;
+}
+
+export function openAiUserFromLocals(locals: Record<string, any>): string {
```

Comment:
Let's change this function so that it is passed a `authn_user_id: string` directly. We are trying to eliminate the use of `locals` as much as possible and we don't want to pass `locals` into library code.

Reply:
Done in ff0b8fae364583747e838607ffc674c57e963e85.